yb-voyager/cmd/exportSchema.go

```diff
@@ -320,35 +314,48 @@ func applyShardedTablesRecommendation(shardedTables []string) (err error) {
 }
 
 /*
-applyShardingRecommendation uses pg_query module to parse the given SQL stmt
+applyShardingRecommendationIfMatching uses pg_query module to parse the given SQL stmt
 In case of any errors or unexpected behaviour it return the original DDL
 so in worse only recommendation of that table won't be followed.
 
 # It can handle cases like multiple options in WITH clause
 
+returns:
+modifiedSqlStmt: original stmt if not sharded else modified stmt with colocation clause
+match: true if its a sharded table and should be modified
+error: nil/non-nil
+
 Drawback: pg_query module doesn't have functionality to format the query after parsing
 so the CREATE TABLE for sharding recommended tables will be one-liner
 */
-func applyShardingRecommendation(sqlInfo *sqlInfo) (string, error) {
+func applyShardingRecommendationIfMatching(sqlInfo *sqlInfo, shardedTables []string) (string, bool, error) {
 	stmt := sqlInfo.stmt
+	formattedStmt := sqlInfo.formattedStmt
 	parseTree, err := pg_query.Parse(stmt)
 	if err != nil {
-		return stmt, fmt.Errorf("error parsing the stmt-%s: %v", stmt, err)
+		return formattedStmt, false, fmt.Errorf("error parsing the stmt-%s: %v", stmt, err)
 	}
 
 	if len(parseTree.Stmts) == 0 {
 		log.Warnf("parse tree is empty for stmt=%s for table '%s'", stmt, sqlInfo.objName)
-		return stmt, nil
+		return formattedStmt, false, nil
 	}
 
 	// Access the first statement directly
 	createStmtNode, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
 	if !ok { // return the original sql if it's not a CreateStmt
 		log.Infof("stmt=%s is not createTable as per the parse tree, expected tablename=%s", stmt, sqlInfo.objName)
-		return stmt, nil
+		return formattedStmt, false, nil
 	}
 	createTableStmt := createStmtNode.CreateStmt
 
+	// Extract schema and table name
+	relation := createTableStmt.Relation
+	parsedTableName := relation.Schemaname + "." + relation.Relname
+	if !slices.Contains(shardedTables, parsedTableName) {
```

Comment:
so, table names in `shardedTables` is always expected to be fully qualified, non-quoted?

Reply:
yeah, for postgresql that is fair expectation i think? What do you say

Reply:
quotes are needed while creating the objects. For the comparisons we can compare unquoted?

Reply:
Yeah comparing unquoted seems okay! I was just confirming 👍

Reply:
yeah cool

Reply:
Just to confirm, as I am not very familiar with the library yet, Is the expectation from `createTableStmt.Relation` also to have an unquoted name always?

Reply:
Yeah that what i observed with this.